envi/config.py

```diff
@@ -80,11 +104,62 @@ def getOptionDoc(self, optname):
         '''
         return self.cfgdocs.get(optname)
 
+    def getConfigPaths(self):
+        '''
+        Return a list of tuples including: (type, valid path strings, existing value)
+
+        'type' can be CONFIG_PATH or CONFIG_ENTRY to indicate whether the tuple 
+        represents a subconfig or an actual key/value pair
+        '''
+        paths = []
+        todo = [ ([], self) ]
+
+        while todo:
+            path, config = todo.pop()
+            
+            if config.keys():
+                pathstr = '.'.join(path) + "."
+                newpaths = [ (CONFIG_ENTRY, "%s%s" % (pathstr, key), "%s" % (config[key]))  for key in config.keys()]
+                paths.extend(newpaths)
+
```

Comment:
Perf nit ( which is not needed here, but is good habit to spot/fix ).  Multiple calls to config.keys() should be replaced by assigning it to a variable:
```
cfgkeys = config.keys()
if cfgkeys:
    dostuff(cfgkeys)
```